web/annotation_layer_builder.js

```diff
@@ -195,6 +215,37 @@ class AnnotationLayerBuilder {
     return !!this.annotationLayer?.hasEditableAnnotations();
   }
 
+  /**
+   * @param {InjectLinkAnnotationsOptions} options
+   * @returns {Promise<void>} A promise that is resolved when the inferred links
+   *   are added to the annotation layer.
+   */
+  async injectLinkAnnotations({
+    inferredLinks,
+    viewport,
+    structTreeLayer = null,
+  }) {
+    if (this._cancelled) {
+      return;
+    }
+
+    const newLinks = this.#annotations
```

Comment:
*Given the previous comment.*
```suggestion
    const newLinks = this.#annotations.length
```